src/main/java/org/jasig/portlet/proxy/service/web/MultiRequestHttpClientServiceImpl.java

```diff
@@ -90,37 +79,46 @@ public AbstractHttpClient getHttpClient(PortletRequest request) {
-         * ie. http://www.google.com:81
-         */
         int httpClientConnectionTimeout = Integer.parseInt(prefs.getValue(HTTP_CLIENT_CONNECTION_TIMEOUT, String.valueOf(DEFAULT_HTTP_CLIENT_CONNECTION_TIMEOUT)));
-        /*
-         * Suggested way of testing Socket Timeout is by using a tool locally to connect
-         * but not respond.  Example tool: bane
-         * http://blog.danielwellman.com/2010/09/introducing-bane-a-test-harness-for-server-connections.html
-         * usage: $bane 10010 NeverRespond
-         * ie. http://localhost:10010
-         */
         int httpClientSocketTimeout = Integer.parseInt(prefs.getValue(HTTP_CLIENT_SOCKET_TIMEOUT, String.valueOf(DEFAULT_HTTP_CLIENT_SOCKET_TIMEOUT)));
-        HttpConnectionParams.setConnectionTimeout(params, httpClientConnectionTimeout);
-        HttpConnectionParams.setSoTimeout(params, httpClientSocketTimeout);
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(httpClientConnectionTimeout)
+                //.setConnectionRequestTimeout(httpClientConnectionTimeout)
+                .setSocketTimeout(httpClientSocketTimeout).build();
+        final HttpClient client = HttpClientBuilder.create()
+                .setDefaultRequestConfig(config)
+                .addInterceptorFirst(new RedirectTrackingResponseInterceptor())
+                .build();
         return client;
     }
+
+//    private HttpClient setHttpClientTimeouts(PortletRequest request, HttpClient client) {
+//        PortletPreferences prefs = request.getPreferences();
+//        HttpParams params = client.getParams();
+//        if (params == null) {
+//            params = new BasicHttpParams();
+//            client.setParams(params);
+//        }
+//        /*
+//         * The connection is attempted 5 times prior to stopping
+//         * so the actual time before failure will be 5 times this setting.
+//         * Suggested way of testing Connection Timeout is by hitting a
+//         * domain with a port that is firewalled:
+//         * ie. http://www.google.com:81
+//         */
+//        int httpClientConnectionTimeout = Integer.parseInt(prefs.getValue(HTTP_CLIENT_CONNECTION_TIMEOUT, String.valueOf(DEFAULT_HTTP_CLIENT_CONNECTION_TIMEOUT)));
+//        /*
+//         * Suggested way of testing Socket Timeout is by using a tool locally to connect
+//         * but not respond.  Example tool: bane
+//         * http://blog.danielwellman.com/2010/09/introducing-bane-a-test-harness-for-server-connections.html
+//         * usage: $bane 10010 NeverRespond
+//         * ie. http://localhost:10010
+//         */
+//        int httpClientSocketTimeout = Integer.parseInt(prefs.getValue(HTTP_CLIENT_SOCKET_TIMEOUT, String.valueOf(DEFAULT_HTTP_CLIENT_SOCKET_TIMEOUT)));
+//        HttpConnectionParams.setConnectionTimeout(params, httpClientConnectionTimeout);
+//        HttpConnectionParams.setSoTimeout(params, httpClientSocketTimeout);
+//        return client;
+//    }
```

Comment:
@timlevett : this was cleaned, please have a look to let me know if something remains